src/XIVLauncher.Common.Unix/Compatibility/WineRunner.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Serilog;
+using XIVLauncher.Common.Util;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public class WineRunner
+{
+    public string RunCommand { get; }
+
+    public string RunArguments { get; }
+
+    public string WineServer { get; }
+
+    public string PathArguments { get; }
+
+    private string wineFolder { get; }
+
+    private string downloadUrl;
+
+    private string toolFolder;
+
+    private string xlcoreFolder;
+
+    public Dictionary<string, string> Environment { get; }
+
+    public WineRunner(string runCmd, string runArgs, string folder, string url, string rootFolder, Dictionary<string, string> env = null, bool is64only = false)
```

Comment:
`is64only` is a misnomer here, in general all wine builds with 64 bit support will have a dedicated `wine64` executable which is the one that should be invoked for xiv and Dalamud,  as when you use the wrong 32 bit one, wine will notice and call wine64 for you (except for some IL-only executables).

`is64only` should always be true, if anything, but is not even needed to be specified at all... The only exception to not calling wine64 is using the new experimental wine8 wow64 backend, which will create a single 64 bit only `wine` executable, that internally thunks all system calls when passing it a 32 bit PE executable (which ironically means when you have a true 64 bit only wine 8 `is64only` should be false here).

This is easy enough to handle transparently handle though, always pick `wine64` unless it doesn't exist, then pick `wine`.